tests/unit/sagemaker/serve/builder/test_hf_dlc_builder.py

```diff
@@ -0,0 +1,74 @@
+# Copyright Amazon.com, Inc. or its affiliates. All Rights Reserved.
+#
+# Licensed under the Apache License, Version 2.0 (the "License"). You
+# may not use this file except in compliance with the License. A copy of
+# the License is located at
+#
+#     http://aws.amazon.com/apache2.0/
+#
+# or in the "license" file accompanying this file. This file is
+# distributed on an "AS IS" BASIS, WITHOUT WARRANTIES OR CONDITIONS OF
+# ANY KIND, either express or implied. See the License for the specific
+# language governing permissions and limitations under the License.
+from __future__ import absolute_import
+from unittest.mock import MagicMock, patch
+
+import unittest
+from sagemaker.serve.builder.model_builder import ModelBuilder
+from sagemaker.serve.mode.function_pointers import Mode
+from sagemaker.serve import ModelServer
+
+from sagemaker.serve.utils.predictors import HfDLCLocalModePredictor
+
+mock_model_id = "bert-base-uncased"
+mock_prompt = "The man worked as a [MASK]."
+mock_sample_input = {"inputs": mock_prompt}
+mock_sample_output = [{'score': 0.0974755585193634, 'token': 10533, 'token_str': 'carpenter', 'sequence': 'the man worked as a carpenter.'},
+                      {'score': 0.052383411675691605, 'token': 15610, 'token_str': 'waiter', 'sequence': 'the man worked as a waiter.'},
+                      {'score': 0.04962712526321411, 'token': 13362, 'token_str': 'barber', 'sequence': 'the man worked as a barber.'},
+                      {'score': 0.0378861166536808, 'token': 15893, 'token_str': 'mechanic', 'sequence': 'the man worked as a mechanic.'},
+                      {'score': 0.037680838257074356, 'token': 18968, 'token_str': 'salesman', 'sequence': 'the man worked as a salesman.'}]
+mock_schema_builder = MagicMock()
+mock_schema_builder.sample_input = mock_sample_input
+mock_schema_builder.sample_output = mock_sample_output
+
+
+class TestHFDlcBuilder(unittest.TestCase):
+
+    @patch("sagemaker.serve.builder.hf_dlc_builder._capture_telemetry", side_effect=None)
+    @patch("sagemaker.serve.builder.hf_dlc_builder._get_ram_usage_mb", return_value=1024)
+    @patch("sagemaker.serve.builder.hf_dlc_builder._get_nb_instance", return_value="ml.g5.24xlarge")
+    def test_build_deploy_for_hf_dlc_local_container(
+            self,
+            mock_get_nb_instance,
+            mock_get_ram_usage_mb,
+            mock_telemetry,
+    ):
+        builder = ModelBuilder(
+            model=mock_model_id,
+            schema_builder=mock_schema_builder,
+            mode=Mode.LOCAL_CONTAINER,
+            model_server=ModelServer.HUGGINGFACE_DLC,
```

Comment:
Can we add test case where customer doesn't need to provide the model_server, and so verify that constructor could detect the model ID and metadata to assign default model server and DLC accordingly. When customer provide the model server input (e.g., TGI, DJL_SERVING, MMS), it will use customer specified value to override the defaults.

Reply:
Can do

Reply:
Addressed